tests/eval/eval_test.go

```diff
@@ -0,0 +1,363 @@
+		entries = append(entries, Entry(tc, tc))
+	}
+	return entries
+}
+
+func loadIntegrationEntries() []TableEntry {
+	cwd, err := os.Getwd()
+	Expect(err).NotTo(HaveOccurred())
+
+	integrationPath := filepath.Join(cwd, testdataDir, integrationDir)
+	cases, err := discoverTestCases(integrationPath)
+	if err != nil || len(cases) == 0 {
+		return nil
+	}
+
+	var entries []TableEntry
+	for _, tc := range cases {
+		entries = append(entries, Entry(tc, tc))
+	}
+	return entries
+}
+
+type evalResult struct {
+	Pass   bool   `json:"pass"`
+	Reason string `json:"reason"`
+}
+
+func stripMarkdownCodeBlock(s string) string {
+	s = strings.TrimSpace(s)
+	s = strings.TrimPrefix(s, "```json")
+	s = strings.TrimPrefix(s, "```")
+	s = strings.TrimSuffix(s, "```")
+	return strings.TrimSpace(s)
+}
+
+func readAndApplyPatch(patchPath string) {
+	By("reading and applying patch")
+	patchContent, err := os.ReadFile(patchPath)
+	Expect(err).NotTo(HaveOccurred())
+
+	cmd := exec.Command("git", "apply", "-")
+	cmd.Dir = tempDir
+	cmd.Stdin = bytes.NewReader(patchContent)
+	output, err := cmd.CombinedOutput()
+	Expect(err).NotTo(HaveOccurred(), "git apply failed: %s", string(output))
+}
+
+// runAPIReview and runJudge can probably share some common code.
+func runAPIReview(model string) (string, float64) {
+	By(fmt.Sprintf("running API review via Claude (%s)", model))
+	ctx, cancel := context.WithTimeout(context.Background(), claudeTimeout)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, "claude",
+		"--print",
+		"--dangerously-skip-permissions",
+		"--model", model,
+		"-p", "/api-review",
+		"--allowedTools", "Bash,Read,Grep,Glob,Task",
+		"--output-format", "json",
+	)
+	cmd.Dir = tempDir
+
+	output, err := cmd.CombinedOutput()
+	Expect(err).NotTo(HaveOccurred(), "claude command failed: %s", string(output))
+
+	var parsed claudeOutput
+	err = json.Unmarshal(output, &parsed)
+	Expect(err).NotTo(HaveOccurred(), "failed to parse claude output: %s", string(output))
+
+	totalReviewerCost += parsed.TotalCostUSD
+	return parsed.Result, parsed.TotalCostUSD
+}
+
+func runJudge(model, reviewOutput, expectedIssues string) (evalResult, float64) {
+	By(fmt.Sprintf("comparing results with Claude judge (%s)", model))
+	ctx, cancel := context.WithTimeout(context.Background(), claudeTimeout)
+	defer cancel()
+
+	prompt := fmt.Sprintf(judgePromptTemplate, reviewOutput, expectedIssues)
+	cmd := exec.CommandContext(ctx, "claude",
+		"--print",
+		"--dangerously-skip-permissions",
+		"--model", model,
+		"-p", prompt,
+		"--output-format", "json",
+	)
+	cmd.Dir = tempDir
+
+	output, err := cmd.CombinedOutput()
+	Expect(err).NotTo(HaveOccurred(), "claude judge command failed: %s", string(output))
+
+	var parsed claudeOutput
+	err = json.Unmarshal(output, &parsed)
+	Expect(err).NotTo(HaveOccurred(), "failed to parse judge output: %s", string(output))
+
+	totalJudgeCost += parsed.TotalCostUSD
+
+	var result evalResult
+	jsonStr := stripMarkdownCodeBlock(parsed.Result)
```

Comment:
The judge isn't good at obeying the instructions not to include extra formatting?